script/bootstrap

```diff
@@ -54,10 +57,17 @@ install_packages_apt()
     sudo apt-get install --no-install-recommends -y rsyslog
 
     # For DBus server
-    sudo apt-get install --no-install-recommends -y libdbus-1-dev
+    sudo apt-get install --no-install-recommends -y dbus libdbus-1-dev
 
     # mDNS
-    sudo apt-get install --no-install-recommends -y libavahi-client3 libavahi-common-dev libavahi-client-dev avahi-daemon
+    sudo apt-get install --no-install-recommends -y libavahi-client3 libavahi-common-dev libavahi-client-dev
+
+    # Thread Certification tests require Avahi to publish records for tests. Since the
+    # same image is used for all tests this needs to be installed here.
+    if [[ ${OTBR_MDNS} == "avahi" || ${OT_BACKBONE_CI} == 1 ]]; then
```

Comment:
@sunytt Do you think `${OT_BACKBONE_CI} == 1` will be true when building a BR reference release? Is there a more accurate variable?

Reply:
I've modified the PR to ensure Avahi is installed if `REFERENCE_DEVICE=1`.